File: analyze_editabletitle.js
const fs = require('fs');
const path = require('path');

// Read the HTML file
const filePath = path.join(__dirname, 'views', 'reportdetails.html');
const content = fs.readFileSync(filePath, 'utf-8');

// Find all occurrences of editableTitle
const regex = /class="[^"]*editableTitle[^"]*"/g;
let match;
const matches = [];

while ((match = regex.exec(content)) !== null) {
    matches.push({
        index: match.index,
        text: match[0]
    });
}

console.log(`Found ${matches.length} occurrences of 'editableTitle'`);
console.log('\n' + '='.repeat(80) + '\n');

// For each match, check if it's wrapped in a div with class="shiny"
matches.forEach((match, i) => {
    const beforeContext = content.substring(Math.max(0, match.index - 1000), match.index);
    const afterContext = content.substring(match.index, Math.min(content.length, match.index + 500));
    
    // Find the full input element
    const inputMatch = afterContext.match(/^[^>]*>/);
    const fullInput = inputMatch ? match.text + inputMatch[0] : match.text;
    
    // Check for shiny wrapper
    const shinyRegex = /<div[^>]*class="[^"]*shiny[^"]*"[^>]*>/g;
    let shinyMatch;
    let lastShinyIndex = -1;
    
    while ((shinyMatch = shinyRegex.exec(beforeContext)) !== null) {
        lastShinyIndex = shinyMatch.index;
    }
    
    let hasShiny = false;
    if (lastShinyIndex !== -1) {
        // Count divs between shiny and editableTitle
        const betweenContent = beforeContext.substring(lastShinyIndex);
        const divOpens = (betweenContent.match(/<div[^>]*>/g) || []).length;
        const divCloses = (betweenContent.match(/<\/div>/g) || []).length;
        
        if (divOpens > divCloses) {
            hasShiny = true;
        }
    }
    
    // Extract section information
    const placeholderMatch = afterContext.match(/placeholder="([^"]*)"/);
    const valueMatch = afterContext.match(/value="([^"]*)"/);
    const idMatch = afterContext.match(/id="([^"]*)"/);
    
    console.log(`Occurrence ${i + 1}:`);
    console.log(`Has shiny wrapper: ${hasShiny ? 'YES ✓' : 'NO ✗'}`);
    
    if (placeholderMatch) {
        console.log(`Placeholder: "${placeholderMatch[1]}"`);
    }
    if (valueMatch) {
        console.log(`Value: "${valueMatch[1]}"`);
    }
    if (idMatch) {
        console.log(`ID: "${idMatch[1]}"`);
    }
    
    // Get surrounding context to identify section
    const lineStart = content.lastIndexOf('\n', match.index) + 1;
    const lineEnd = content.indexOf('\n', match.index + 100);
    const line = content.substring(lineStart, lineEnd > -1 ? lineEnd : content.length);
    console.log(`Line preview: ${line.substring(0, 100).trim()}...`);
    
    console.log('\n' + '-'.repeat(80) + '\n');
});

// Summary
const withShiny = matches.filter((match, i) => {
    const beforeContext = content.substring(Math.max(0, match.index - 1000), match.index);
    const shinyRegex = /<div[^>]*class="[^"]*shiny[^"]*"[^>]*>/g;
    let shinyMatch;
    let lastShinyIndex = -1;
    
    while ((shinyMatch = shinyRegex.exec(beforeContext)) !== null) {
        lastShinyIndex = shinyMatch.index;
    }
    
    if (lastShinyIndex !== -1) {
        const betweenContent = beforeContext.substring(lastShinyIndex);
        const divOpens = (betweenContent.match(/<div[^>]*>/g) || []).length;
        const divCloses = (betweenContent.match(/<\/div>/g) || []).length;
        return divOpens > divCloses;
    }
    return false;
}).length;

console.log('\nSUMMARY:');
console.log(`Total editableTitle inputs: ${matches.length}`);
console.log(`With shiny wrapper: ${withShiny}`);
console.log(`Missing shiny wrapper: ${matches.length - withShiny}`);

File: analyze_editabletitle.py
import re

# Read the HTML file
with open('views/reportdetails.html', 'r', encoding='utf-8') as f:
    content = f.read()

# Find all occurrences of editableTitle with context
pattern = r'(?:.*\n)?.*?(?:class="[^"]*editableTitle[^"]*"|editableTitle).*?(?:\n.*)?'
matches = list(re.finditer(pattern, content, re.MULTILINE))

print(f"Found {len(matches)} occurrences of 'editableTitle'")
print("\n" + "="*80 + "\n")

# For each match, check if it's wrapped in a div with class="shiny"
for i, match in enumerate(matches):
    start = match.start()
    end = match.end()
    
    # Get context before the match (up to 500 chars)
    context_start = max(0, start - 500)
    context_before = content[context_start:start]
    
    # Get the match itself
    match_text = match.group(0).strip()
    
    # Check if there's a div with class="shiny" before this match
    shiny_pattern = r'<div[^>]*class="[^"]*shiny[^"]*"[^>]*>'
    shiny_matches = list(re.finditer(shiny_pattern, context_before))
    
    # Check if the last shiny div is still open (not closed)
    has_shiny = False
    if shiny_matches:
        last_shiny = shiny_matches[-1]
        # Count divs opened and closed after the shiny div
        after_shiny = context_before[last_shiny.end():]
        div_opens = len(re.findall(r'<div[^>]*>', after_shiny))
        div_closes = len(re.findall(r'</div>', after_shiny))
        if div_opens >= div_closes:
            has_shiny = True
    
    print(f"Occurrence {i+1}:")
    print(f"Line content: {match_text}")
    print(f"Has shiny wrapper: {'YES' if has_shiny else 'NO'}")
    
    # Get more context to identify the section
    line_start = content.rfind('\n', 0, start) + 1
    line_end = content.find('\n', end)
    if line_end == -1:
        line_end = len(content)
    
    # Look for nearby text that might indicate the section
    section_context = content[max(0, line_start-200):min(len(content), line_end+200)]
    
    # Extract any placeholder or value from the input
    input_match = re.search(r'placeholder="([^"]*)"', section_context)
    if not input_match:
        input_match = re.search(r'value="([^"]*)"', section_context)
    
    if input_match:
        print(f"Section identifier: {input_match.group(1)}")
    
    print("\n" + "-"*80 + "\n")